src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Figure defaultFigure = new Circle(Color.WHITE, 10);
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final double maxValue = 10;
+    private final int amountOfFigures = 5;
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(1, amountOfFigures + 1);
+        return switch (randomNumber) {
+            case 1 -> new Circle(
+                    colorSupplier.getRandomColor(),
+                    random.nextDouble(maxValue)
+            );
+            case 2 -> new Square(
+                    colorSupplier.getRandomColor(),
+                    random.nextDouble(maxValue)
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create separate variables for the color and the side, and then pass them to the constructor for better code readability.